src/main/java/com/slack/eithernet/Retries.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2023 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.eithernet
+
+import kotlin.random.Random
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.delay
+
+/**
+ * Retries a [block] of code with exponential backoff.
+ *
+ * @param maxAttempts The maximum number of times to retry the operation. Default is 5.
+ * @param initialDelay The delay before the first retry. Default is 1 second.
+ * @param delayFactor The factor by which the delay should increase after each failed attempt.
+ *   Default is 2.0.
+ * @param maxDelay The maximum delay between retries. Default is [Duration.INFINITE].
+ * @param jitterFactor The maximum factor of jitter to introduce. For example, a value of 0.1 will
+ *   introduce up to 10% jitter. Default is 0.
+ * @param onFailure An optional callback for failures, useful for logging.
+ * @param block The block of code to retry. This block should return an [ApiResult].
+ * @return The result of the operation if it's successful, or the last failure result if all
+ *   attempts fail.
+ *
+ * This function will attempt the operation you give it up to [maxAttempts] times, multiplying the
```

Comment:
weird for params to not be at the end

Reply:
Derp [0e1e365](https://github.com/slackhq/EitherNet/pull/60/commits/0e1e3659133fd7e1703cdf5433ff807c66e36cd3)